### PR TITLE
NIAD-276 FHIR bundle creation mechanism

### PR DESCRIPTION
**Pull request by dawidskainos**

### CODE REVIEW

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/EncounterReportBundleService.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+        addEncounter(bundle, encounter);
+        addEpisodeOfCare(bundle, encounter);
+
+        return bundle;
+    }
+
+    private void addEncounter(Bundle bundle, Encounter encounter) {
+        bundle.addEntry()
+            .setFullUrl(encounter.getIdElement().getValue())
```
Comment (Alicja-Kainos) [inline service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/EncounterReportBundleService.java:37]: IdElement is random UUID -> why is it set here as url?

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/service/ReportService.java [hunk 2]
```diff
@@ -1,37 +1,40 @@
 package uk.nhs.adaptors.oneoneone.cda.report.service;
```
Comment (Alicja-Kainos) [inline service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/service/ReportService.java:1]: Shouldn't we rename this to EncounterService?

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/EncounterReportBundleService.java (hunk 1 above)
Comment (dawidskainos) [reply service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/EncounterReportBundleService.java:37]: URL is a "resource locator" so UUID identifies a resource quite well within a bundle message.
See: https://hapifhir.io/hapi-fhir/docs/client/examples.html

Comment (dawidskainos) [reply service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/EncounterReportBundleService.java:37]: Good idea. Done.

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/service/ReportService.java (hunk 2 above)
Comment (dawidskainos) [reply service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/service/ReportService.java:1]: Good idea. Done.

**Pull request merged**